presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/HistoryTable.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.InMemoryRecordSet;
+import io.prestosql.spi.connector.RecordCursor;
+import io.prestosql.spi.connector.SchemaTableName;
+import io.prestosql.spi.connector.SystemTable;
+import io.prestosql.spi.predicate.TupleDomain;
+import org.apache.iceberg.HistoryEntry;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.util.SnapshotUtil;
+
+import java.util.List;
+import java.util.Set;
+
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.BooleanType.BOOLEAN;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
+import static java.util.Objects.requireNonNull;
+
+public class HistoryTable
+        implements SystemTable
+{
+    private final SchemaTableName tableName;
+    private final Table icebergTable;
+
+    private static final List<ColumnMetadata> COLUMNS = new ImmutableList.Builder<ColumnMetadata>().add(
+            new ColumnMetadata("made_current_at", TIMESTAMP),
+            new ColumnMetadata("snapshot_id", BIGINT),
+            new ColumnMetadata("parent_id", BIGINT),
+            new ColumnMetadata("is_current_ancestor", BOOLEAN)).build();
+
+    public HistoryTable(SchemaTableName tableName, Table icebergTable)
+    {
+        this.tableName = requireNonNull(tableName, "tableName is null");
```

Comment:
`tableName` is used to construct `ConnectorTableMetadata` only. Will it be better to instantiate `ConnectorTableMetadata` in the `HistoryTable` constructor once?

Reply:
Thanks, updated.